### PR TITLE
added logic to detect if a script is being loaded and it already exis…

### PR DESCRIPTION
**Pull request by MiguelCastillo**
…ts in the DOM

https://github.com/MiguelCastillo/load-js/issues/8

### CODE REVIEW

**Comment by MiguelCastillo** [comment]
The logic in this PR adds the ability to detect if the script being loaded already exists in the DOM by ID or by URL. This is useful in situation where script tags are being loaded outside of loadJS and it, and loadJS still needs to gracefully handle it.

The logic introduced will basically make use of the scripts already loaded as if it had been loaded by loadJS, unless the `allowExternal` option is set to false. When the option `allowExternal` is set to false, loadJS will load a new script.

**Comment by MiguelCastillo** [comment]
Ok, so there really isn't any difference in your code with what I have.  Line 23 sets `var id = options.id || options.url;` .  So if there is an `el` in the DOM, it is coming from either the `options.id` or the `options.url`. There really isn't a case in the logic that the `if (id)` you have introduced will be false.

Or am I missing something more subtle?

Thanks for taking a look!!

File: index.js [hunk 1]
```diff
@@ -20,36 +21,27 @@
     }
 
     var id = options.id || options.url;
```
Comment (MiguelCastillo) [inline index.js]: here we set the id used in line 32.

**Comment by MiguelCastillo** [comment]
I tweaked the logic and the variable names to try to make the code a bit more clear.

**Comment by allpro** [comment]
You are correct. I missed that the id-var could also be options.url. I read id as options.id.

**Comment by MiguelCastillo** [comment]
I follow what you are saying. But there is no logic checking for `if (options.id)`.  The logic is checking for `if (id)`, and that `id` is actually the `cacheId`. The `cacheId` is derived from `options.id || options.url`.  I renamed `id` to `cacheId` to avoid confusion with `option.id`.

The logic is basically reduced to this:

```
var cacheId = options.id || options.url;
var el = getScriptById(options.id) || getScriptByUrl(options.url);

if (el) {
   /// there must a calculated cacheId.
   assert(cacheId);
}
```

If there is an el then there is a `cacheId`.  The only edge is if the ID is `0` but that's a minor defect we can ignore for now.

**Comment by MiguelCastillo** [comment]
@allpro sorry i responded again.  for whatever reason I didnt see your last response when i was replying. Im glad I am not going crazy - I thought I was :D  thanks a bunch for looking at this!

**Comment by allpro** [comment]
I realized my mistake as soon as I posted and corrected my comment.

Changing it to `cacheId` is a good idea so others won't make the same dumb mistake as me in the future.

I really shouldn't try to read and comment on code on my phone! ;)

Keep up the great work.

**Pull request merged**